core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -385,8 +452,12 @@ class DynamicBrokerConfig(private val kafkaConfig: KafkaConfig) extends Logging
     allNewConfigs.asScala.foreach { case (k, v) => newConfigs.put(k, v.asInstanceOf[AnyRef]) }
     newConfigs.putAll(newCustomConfigs)
     if (validateOnly) {
-      if (!reconfigurable.validateReconfiguration(newConfigs))
-        throw new ConfigException("Validation of dynamic config update failed")
+      try {
+        reconfigurable.validateReconfiguration(newConfigs)
+      } catch {
+        case e: ConfigException => throw e
+        case _: Exception => throw new ConfigException("Validation of dynamic config update failed with class " + reconfigurable.getClass)
+      }
     } else
       reconfigurable.reconfigure(newConfigs)
```

Comment:
Maybe we could have an info log message which mentioned the reconfigurable and the custom configs that are being updated?